bbp/default.nix

```diff
@@ -415,6 +415,12 @@ let
             numpy = pythonPackages.numpy;
         };
 
+        flatindexer-py3 = callPackage ./hpc/FLATIndexer {
```

Comment:
recalling callPackage is generally not the way to proceed, because it causes a re-interpretation.

Could you try something like flatindexer-py3 = flatindexer.override { python = python3; } 

?